java/connectors/semantickernel-connectors-memory-azurecognitivesearch/src/main/java/com/microsoft/semantickernel/connectors/memory/azurecognitivesearch/AzureCognitiveSearchMemoryStore.java

```diff
@@ -54,6 +56,7 @@ public class AzureCognitiveSearchMemoryStore implements MemoryStore {
     private final SearchIndexAsyncClient _adminClient;
 
     private final Map<String, SearchAsyncClient> _clientsByIndex = new ConcurrentHashMap<>();
+    private final Function<SearchDocument, MemoryRecord> memoryRecordMapper;
```

Comment:
Just a thought... if this was a BiFunction<SearchDocument, MemoryRecord, Boolean>, then you could default the memoryRecordMapper with  AzureCognitiveSearchMemoryRecord::toMemoryRecord and do away with the if block in getNearestMatchesAsync. But, as a supplier of the Function<SearchDocument, MemoryRecord> I can include or exclude embeddings, so it would only help to make a default. So, nevermind... 

Would it be better to pass the mapper in the call to getNearestMatch(es)Async? Maybe how you map the SearchDocument depends on the collection or some other factor. If so, you'd have to create a new AzCSMS for each unique handling of a collection.

Reply:
adding it on the getNearestMatchesAsync would mean adding it to the MemoryStore interface, that would then mean leaking some ACS specific classes onto the MemoryStore interface, I am sort of ok with doing specific stuff in a constructor, but not so much on the interface